### DISCUSSION TITLE
Node set id and Suface (Side set) id overlap in BCs on an Abaqus .inp mesh

### DISCUSSION THREAD

**Discussion by bylore** [idaholab/moose#20036] [Q&A Meshing] [answered]
hi everyone
I am new to moose. Recently, i tried to solve very simple linear elastic problems using moose tensor mechanics modules. 
My mesh is built from an abaqus inp file. The boundary part commands are

![image](https://user-images.githubusercontent.com/18157950/149293727-4f44e81d-8d4b-443c-ac8e-5d911a997eb9.png)

Case 1
mesh contains:
![22c16aa0b7e9302d62d3ffb20b3cd30](https://user-images.githubusercontent.com/18157950/149288796-9f560b88-6004-4b3c-8aa2-76c07f81969e.png)
the results are
![image](https://user-images.githubusercontent.com/18157950/149289134-25db7de3-540b-40e3-8843-fd591bed75a9.png)
![image](https://user-images.githubusercontent.com/18157950/149289075-ca04a06e-6d12-45e5-9444-17a3468ecdcf.png)

Case2
mesh contains:
![1642060805(1)](https://user-images.githubusercontent.com/18157950/149289373-d045ba95-114f-4371-8631-94cc370b0447.png)
the results are
![image](https://user-images.githubusercontent.com/18157950/149289816-93dc58cd-8a94-4415-ae70-1649003bc7f4.png)
![image](https://user-images.githubusercontent.com/18157950/149289851-5d4f6a62-e591-4501-ac8d-1df985c993be.png)

Case3
mesh contains
![image](https://user-images.githubusercontent.com/18157950/149292740-cc6d7557-a4f7-4119-a056-9bfe495beb25.png)
the results are
![image](https://user-images.githubusercontent.com/18157950/149293028-63aa83c3-ea98-4270-b511-d8086fc532b2.png)
![image](https://user-images.githubusercontent.com/18157950/149292997-86487257-5c82-4104-8355-bbdbdcd6c26e.png)

I guess Moose gives ids to Sets and Surface respectively according to their names, but the BCs are applied without distinguishing Sets boundaries (node sets) or Surface boundaries (side sets).
in Case1, it is,
Sets:  bottom = 0; left = 1; right = 2; top = 3
Surfaces: s_bottom = 0; s_left = 1; s_right = 2; s_top = 3

in Case2, it is:
Sets: all = 0; bottom = 1; left = 2; right = 3; top = 4
Surfaces: s_bottom = 0; s_left = 1; s_right = 2; s_top = 3

in Case3, it is:
Sets: all = 0; bottom = 1; left = 2; right = 3; top = 4
Surfaces: s_a0 = 0; s_a1 = 1; s_a2 = 2; s_a3 = 3; s_a4 = 4;  (they are all bottom side of the squre)
s_bottom = 5; s_left = 6; s_right = 7; s_top = 8

So everything goes well in Case1 because Sets and Surfaces have coincident ids，but in Case2, The boundary on "bottom" will be applied to both "bottom" and "s_left"  because they have same id  and the boundary on "s_top" is applied to both "right" and 's_top'.
Case3 works well again as expected.

The software information is 
MOOSE Version:           git commit dbb9945 on 2021-12-20
LibMesh Version:         5f606f4dbd542262512cf701456d1ab7cbd65111
PETSc Version:           3.15.1
SLEPc Version:           3.15.1

This issue can be avoid by adjusting the mesh file. I just report this here. or you can ignore this if it's because I'm using moose in some inappropriate way.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Yes, MOOSE does this by default, for simplicity. All nodes from a given sideset are given the nodeset ID that is the same as the sideset ID (or equivalently, a nodeset with the same ID is created for every sideset), as explained here https://mooseframework.inl.gov/source/mesh/MooseMesh.html

This is done at the end of the mesh generation. It's unfortunate that this ends up changing the result for your case since it makes the BC apply on more nodes than wanted. To avoid this, you can rename sidesets using a RenameBoundaryGenerator, in the [Mesh] block
https://mooseframework.inl.gov/source/meshgenerators/RenameBoundaryGenerator.html

Guillaume
